### DISCUSSION TITLE
Mirage-Layout Template Support Jakarta?

### DISCUSSION THREAD

**Discussion by adilaltun** [primefaces/community#2967] [PrimeFaces Templates] [answered]
Hello, 

Does Mirage-Layout template support Jakarta version?

**Comment by melloware** [comment, marked as answer]
Yes. Mine does.  Make sure to download the latest from your PrimeStore account.

**Reply by adilaltun** [reply]
Hello, thank you for your quick response.

Primefaces Mirage-Layout template works flawlessly with **Tomcat 9, Java 17 and primefaces-10.0.0.jar.** You can see the screenshot of the web browser sources tab of this topic below.

![running-program_1](https://github.com/user-attachments/assets/af19d62a-387c-428a-a4da-fd502cf3e5cb)

However, when the project is **upgraded to Tomcat 10 and tried to run with primefaces-10.0.0-jakarta.jar**, you can see the screenshot of the web browser sources tab below.

![webapp3-jakarta-faces](https://github.com/user-attachments/assets/4ffd798c-d1bd-4aa9-a68b-0e4a4cd345da)

Looking at this update, my project, which has been upgraded to Tomcat 10, cannot access the files under resources along with jakarta.faces.resources. What or what could be the reasons for this?

**Reply by melloware** [reply]
What are the errors on your F12 console?  It feels like you are missing something?

**Reply by adilaltun** [reply]
Hello, first of all, thank you again for your quick return.

You can see the screenshot of the browser console tab below.

![console](https://github.com/user-attachments/assets/2815a5a5-c528-4335-997a-d69b9fce9b31)

Additionally, while there is no data in the Network tab when the project is first loaded, I can view the following data in the network tab when the page is refreshed with F5. (Status values ​​of CSS files in the project running in Tomcat 9 are 304)

![network](https://github.com/user-attachments/assets/edac2225-ca80-448d-b5e3-b545e7e3f1a7)

**Reply by melloware** [reply]
OK you are getting a "strict mime type" error so it looks like Tomcat 10 has some strict setting on that was not on in Tomcat 9.  So its not loading your JSS and CSS so something in your Tomcat setup in 10 is different?

**Reply by adilaltun** [reply]
Hello, 

For the project, all javax jars were updated to Jakarta. (This includes the primefaces jar.) Regarding the migration from Tomcat 9 to Tomcat 10, have any different configuration settings been added regarding mime types?

**Reply by melloware** [reply]
I have a feeling its Tomcat 10 related and not so much PrimeFaces or JSF related?  What version of Mojarra are you using?  Maybe ask this on Stack Overflow tagging Mojarra and Tomcat?
